File: Go/palindromes.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
)

func main() {
	file, err := os.Open("/usr/share/dict/words")
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()
	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		word := scanner.Text()
		if isPalindrome(word) {
			fmt.Println(word, "is a palindrome")
		}
	}
}

func isPalindrome(word string) bool {
	var wl, mid, i int

	wl = len(word)
	mid = wl >> 1
	wl--
	for i = 0; mid > i; i++ {
		if word[i]^word[wl-i] != 0 {
			return false
		}
	}

	return true
}

File: JavaScript/palindromes.js
"use strict"
const fs = require('fs')
 
fs.readFile("/usr/share/dict/words", "utf8", (err, text) => {
    var words = text.split("\n");
    for(var i = 0, len = words.length; i < len; i++) {
      if (words[i].length === 0) continue
      var forwards = words[i].toLowerCase()
      var backwards = forwards.split("").reverse().join("")
      if (forwards === backwards) console.log(words[i] + " is a palindrome")
    }
});
